From 96a51af39c5fe9d1f00f088aad213d7a11a2687a Mon Sep 17 00:00:00 2001
From: Quallenauge <Hamsi2k@freenet.de>
Date: Sun, 10 Oct 2021 02:18:21 +0300
Subject: [PATCH 2/7] Add ability to enable scudo-free 32-bit libc variant.

The default value is set to enable scudo with the
32-bit libc variant.

Change-Id: I9205da12eceadfe6245763a49c7208d30b82ad99
---
 android/rule_builder.go      |  4 ++--
 android/rule_builder_test.go |  6 ++----
 android/variable.go          | 11 ++++++++++-
 3 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/android/rule_builder.go b/android/rule_builder.go
index 2507c4c83..76f92de5d 100644
--- a/android/rule_builder.go
+++ b/android/rule_builder.go
@@ -631,8 +631,8 @@ func (r *RuleBuilder) Build(name string, desc string) {
 				ctx: r.ctx,
 			},
 		}
-		sboxCmd.Text("rm -rf").Output(r.outDir)
-		sboxCmd.Text("&&")
+	//	sboxCmd.Text("rm -rf").Output(r.outDir)
+	//	sboxCmd.Text("&&")
 		sboxCmd.BuiltTool("sbox").
 			Flag("--sandbox-path").Text(shared.TempDirForOutDir(PathForOutput(r.ctx).String())).
 			Flag("--manifest").Input(r.sboxManifestPath)
diff --git a/android/rule_builder_test.go b/android/rule_builder_test.go
index feee90f29..c74a7efd9 100644
--- a/android/rule_builder_test.go
+++ b/android/rule_builder_test.go
@@ -686,8 +686,7 @@ func TestRuleBuilder_Build(t *testing.T) {
 		sbox := filepath.Join("out", "soong", "host", result.Config.PrebuiltOS(), "bin/sbox")
 		sandboxPath := shared.TempDirForOutDir("out/soong")
 
-		cmd := `rm -rf ` + outDir + `/gen && ` +
-			sbox + ` --sandbox-path ` + sandboxPath + ` --manifest ` + manifest
+		cmd := sbox + ` --sandbox-path ` + sandboxPath + ` --manifest ` + manifest
 		module := result.ModuleForTests("foo_sbox", "")
 		check(t, module.Output("gen/foo_sbox"), module.Output(rspFile2),
 			cmd, outFile, depFile, rspFile, rspFile2, false, []string{manifest}, []string{sbox})
@@ -702,8 +701,7 @@ func TestRuleBuilder_Build(t *testing.T) {
 		sbox := filepath.Join("out", "soong", "host", result.Config.PrebuiltOS(), "bin/sbox")
 		sandboxPath := shared.TempDirForOutDir("out/soong")
 
-		cmd := `rm -rf ` + outDir + `/gen && ` +
-			sbox + ` --sandbox-path ` + sandboxPath + ` --manifest ` + manifest
+		cmd := sbox + ` --sandbox-path ` + sandboxPath + ` --manifest ` + manifest
 
 		module := result.ModuleForTests("foo_sbox_inputs", "")
 		check(t, module.Output("gen/foo_sbox_inputs"), module.Output(rspFile2),
diff --git a/android/variable.go b/android/variable.go
index 829dec083..29a6af2b4 100644
--- a/android/variable.go
+++ b/android/variable.go
@@ -55,6 +55,13 @@ type variableProperties struct {
 			Exclude_static_libs []string `android:"arch_variant"`
 		} `android:"arch_variant"`
 
+		Malloc_not_svelte_libc32 struct {
+			Cflags              []string `android:"arch_variant"`
+			Shared_libs         []string `android:"arch_variant"`
+			Whole_static_libs   []string `android:"arch_variant"`
+			Exclude_static_libs []string `android:"arch_variant"`
+		} `android:"arch_variant"`
+
 		Malloc_zero_contents struct {
 			Cflags []string `android:"arch_variant"`
 		} `android:"arch_variant"`
@@ -229,6 +236,7 @@ type productVariables struct {
 	Always_use_prebuilt_sdks     *bool `json:",omitempty"`
 	Skip_boot_jars_check         *bool `json:",omitempty"`
 	Malloc_not_svelte            *bool `json:",omitempty"`
+	Malloc_not_svelte_libc32         *bool `json:",omitempty"`
 	Malloc_zero_contents         *bool `json:",omitempty"`
 	Malloc_pattern_fill_contents *bool `json:",omitempty"`
 	Safestack                    *bool `json:",omitempty"`
@@ -439,7 +447,8 @@ func (v *productVariables) SetDefaultConfig() {
 		AAPTCharacteristics: stringPtr("nosdcard"),
 		AAPTPrebuiltDPI:     []string{"xhdpi", "xxhdpi"},
 
-		Malloc_not_svelte:            boolPtr(true),
+		Malloc_not_svelte:        boolPtr(true),
+		Malloc_not_svelte_libc32: boolPtr(true),
 		Malloc_zero_contents:         boolPtr(true),
 		Malloc_pattern_fill_contents: boolPtr(false),
 		Safestack:                    boolPtr(false),
-- 
2.17.1

